feature/experimental/p4rt/otg_tests/google_discovery_protocol_packetin_test/google_discovery_protocol_packetin_test.go

```diff
@@ -111,32 +112,48 @@ type testArgs struct {
 // programmTableEntry programs or deletes p4rt table entry based on delete flag.
 func programmTableEntry(ctx context.Context, t *testing.T, client *p4rt_client.P4RTClient, packetIO PacketIO, delete bool) error {
 	t.Helper()
-	err := client.Write(&p4_v1.WriteRequest{
+	err := client.Write(&p4v1pb.WriteRequest{
 		DeviceId:   deviceID,
-		ElectionId: &p4_v1.Uint128{High: uint64(0), Low: electionID},
+		ElectionId: &p4v1pb.Uint128{High: uint64(0), Low: electionID},
 		Updates: p4rtutils.ACLWbbIngressTableEntryGet(
 			packetIO.GetTableEntry(delete),
 		),
-		Atomicity: p4_v1.WriteRequest_CONTINUE_ON_ERROR,
+		Atomicity: p4v1pb.WriteRequest_CONTINUE_ON_ERROR,
 	})
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
-// decodePacket decodes L2 header in the packet and returns source and destination MAC and ethernet type.
-func decodePacket(t *testing.T, packetData []byte) (string, string, layers.EthernetType) {
+// decodePacket decodes L2 header in the packet and returns source and destination MAC
+// vlanID, and ethernet type.
+func decodePacket(t *testing.T, packetData []byte) (string, string, uint16, layers.EthernetType) {
```

Comment:
As discussed, please add 3 types of flows for both Packet-in and Packet-out

=== Packet-in scenarios
* GDP with vlanID - punted to p4rt controller.
* GDP without vlanID - punted to p4rt controller.
* Non-GDP Ethernet without vlanID - using FIB

=== Packet-out scenarios
* Inject GDP with vlanID punted out a non trunk interface
* Inject GDP without vlanID punted out a non trunk interface
* Inject Non-GDP Ethernet without vlanID to follow the FIB

Reply:
This testcase seems to evolve beyond original requirements. Just curious if non-GDP ethernet injection is a requirement of the solution?

Reply:
The intent is to emulate production traffic which will be non GDP and may be untagged ethernet.